qudipy/control/voltage_pulses.py

```diff
@@ -0,0 +1,570 @@
+                                    var_pulse=np.full(num_val, delta_g_0))
+        rot_pulse.add_control_variable(var_name= 'V_{ind}'.format(ind = i), 
+                                    var_pulse=np.full(num_val, v_offset))
+    # adding in active qubits variable
+    setattr(rot_pulse, 'active', active_qubits) 
+
+    # values of normalized time and pulse
+    tau = np.linspace(0, 1, num_val)
+    v_pulse = v_unit_shape(tau)
+    # Larmor frequency
+    omega = e / m_e * B_0
+    # Converting theta to radians and shifting it to the range (-pi, pi]
+    theta_rad = np.deg2rad(theta) % (2 * np.pi)
+    if theta_rad > np.pi:
+        theta_rad -= 2 * np.pi
+    # define alpha = theta / 2pi
+    alpha = theta_rad / (2 * np.pi)
+    
+    # evaluating the integral of delta g pulse
+    delta_g_pulse = delta_g_interp(v_unit_shape(tau))
+    delta_g_int = np.trapz(delta_g_pulse, tau) - np.full(num_val, delta_g_0)
+    
+    # finding the pulse length and setting the correct time values for pulses
+    T = 1.
+    
+    if axis == 'Z':
+        # in this case, only the voltage on active qubits changes
+        for i in active_qubits:
+            rot_pulse.ctrl_pulses['delta_g_{ind}'.format(ind = i)] = delta_g_pulse
+            rot_pulse.ctrl_pulses['V_{ind}'.format(ind = i)] = v_pulse
+            
+        # pulse duration
+        T = 2 * theta_rad / (omega * np.abs(delta_g_int))
+        
+    elif axis in ['X', 'Y']:
+        # determine the phase values during the pulse first
+        phis = np.full(num_val, 0.)
+        if axis.upper() == 'Y':
+            phis = np.full(num_val, 90)
+        if theta < 0:
+            phis = phis + 180
+            
+        rot_pulse.add_control_variable("phi", phis)
+        # Making the ESR pulse of the same shape as the g-factor pulse
+        # shifted by delta_g_0
+        # finding the coefficient first
+        
+        A = 1 / np.sqrt( (2 * np.pi / theta_rad) ** 2 - 1)
+        
+        # values of ESR field
+        omega_capital = A * omega * np.abs(delta_g_pulse - delta_g_0) / 2
+        b_rf = m_e / e * omega_capital 
+        
+        rot_pulse.add_control_variable("B_rf", b_rf)
+        
+        # unlike the case of ROTZ, the voltage here changes only on the 
+        # idling qubits so that they return to their initial state
+        # after the pulse
+        for i in idle_qubits:
+            rot_pulse.ctrl_pulses['delta_g_{ind}'.format(ind = i)] = delta_g_pulse
+            rot_pulse.ctrl_pulses['V_{ind}'.format(ind = i)] = v_pulse
+            
+        # pulse duration
+        alpha = theta_rad / (2 * np.pi)
+        T = 4 * np.pi / (omega * np.abs(delta_g_int)) *  np.sqrt(1 - alpha ** 2)
+    
+    else:
+        # *** ARBITRARY ROTATION ***
+        #create variables for Vector Components to speed up run-time
+        n_x = rotation_axis[0]
+        n_y = rotation_axis[1]
+        n_z = rotation_axis[2]
```

Comment:
It's the same thing but with a more concise syntax
```suggestion
        n_x, n_y, n_z = rotation_axis
```